ClangSharp.PInvokeGenerator/Cursors/Decls/CXXMethodDecl.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Diagnostics;
+
+namespace ClangSharp
+{
+    internal class CXXMethodDecl : FunctionDecl
```

Comment:
For example, this follows the type hierarchy described here: https://clang.llvm.org/doxygen/classclang_1_1CXXMethodDecl.html.

This is: `Decl -> NamedDecl -> ValueDecl -> DeclaratorDecl -> FunctionDecl -> CXXMethodDecl`

Reply:
This also means that we can centralize the `ParmDecls` property in `FunctionDecl`, rather then repeating it in `CXXMethod`, `Constructor`, `Destructor`, and `ConversionFunction`